src/wpool_process_callbacks.erl

```diff
@@ -0,0 +1,76 @@
+-module(wpool_process_callbacks).
+
+-behaviour(gen_event).
+
+%% gen_event callbacks
+
+-export([ init/1
+        , handle_event/2
+        , handle_call/2
+        , handle_info/2
+        , code_change/3
+        , terminate/2]).
+
+-export([notify/3]).
+-type state() :: ordset:ordset(module()).
+
+-type event() :: on_init_start | on_new_worker | on_worker_dead.
+
+-spec init([wpool:option()]) -> {ok, state()}.
+init(WPoolOpts) ->
+{ok, maybe_initial_callbacks(WPoolOpts)}.
+
+-spec handle_event({event(), [any()]}, state()) -> {ok, state()}.
+handle_event({Event, Args}, Callbacks) ->
+  [call(Callback, Event, Args) || Callback <- Callbacks],
```

Comment:
If instead of managing a list of callbacks here, we let `gen_event` take care of that, this module (and in particular this function and the one below) will be simplified.